include/exadg/aero_acoustic/user_interface/parameters.h

```diff
@@ -54,6 +56,14 @@ class Parameters
 
     AssertThrow(fluid_to_acoustic_coupling_strategy != FluidToAcousticCouplingStrategy::Undefined,
                 dealii::ExcMessage("Coupling strategy has to be set."));
+
+    if(compute_acoustic_from_analytical_cfd_solution)
+    {
+      AssertThrow(
+        not compute_acoustic_from_analytical_source_term,
+        dealii::ExcMessage(
+          "Only one of the following can be true: compute_acoustic_from_analytical_cfd_solution, compute_acoustic_from_analytical_source_term."));
+    }
```

Comment:
Doesn't this suggest the usage of only a single boolean parameter, or the usage of an enum parameter (e.g. `AcousticSourceTermComputation`) instead (`Patterns::Enum` should allow this for easy use in applications :))?

Reply:
Looking at the rest of the code, I agree we might indeed need two parameter. What about the following set of parameters:

- one boolean parameter `use_analytical_fluid_solution`
- one enum parameter `AcousticSourceTermComputation` with two values like `Analytical` vs. `FromFluidSolution`

I think with these parameters I would understand the logic of the code more easily. According to my current understanding of the code, the combination `use_analytical_fluid_solution = true` and  `AcousticSourceTermComputation::Analytical` should be possible (I would understand this option as using `TimeIntAnalytic` for the fluid problem, but "overwriting/replacing" the usual source term for acoustic by an analytical function). However, I also understand to exclude this option from a user perspective. 

What I do not like regarding the current parameter name `compute_acoustic_from_analytical_cfd_solution` is that this name does not seem to reflect that this parameter also affects the solution of the fluid problem (and the fluid time integrator) and not just the way how the acoustic source term is computed (i.e. a user could understand this parameter as solving the fluid problem as usual but only replacing the simulated fluid solution by the analytical one when it comes to the computation of the source term).

I hope this comment clarifies my thoughts. It might be that I still did not fully understand the code.

Reply:
Moving the time integrator to the IncNSE module made it comparably simple. The proposed version is a mix of your idea and the parameter that is now loaced in IncNSE [5efc732](https://github.com/exadg/exadg/pull/659/commits/5efc7327890f5a84d6d550396244124f94633f33)